libs/features/mas/web-components/src/merch-card.js

```diff
@@ -102,22 +105,35 @@ export class MerchCard extends LitElement {
         ) {
             this.style.border = this.computedBorderStyle;
         }
-        this.updateComplete.then(async () => {
-            const allPrices = Array.from(
-              this.querySelectorAll('span[is="inline-price"][data-wcs-osi]'),
-            );
-            // Filter out prices within the callout-content slot
-            const prices = allPrices.filter(
-                (price) => !price.closest('[slot="callout-content"]'),
-            );
-            await Promise.all(prices.map((price) => price.onceSettled()));
-            this.variantLayout.postCardUpdateHook(this);
-        });
+        if (changedProperties.has('variant')) {
+            this.variantLayout = getVariantLayout(this);
+            this.variantLayout.connectedCallbackHook();
```

Comment:
either we rename that method, either we don't use it in constructor

Reply:
or may be we can move connectedCallbackHook in some internal method of VariantLayout constructor?

Reply:
makes sense, I'll rename to `postUpdatedHook`

Reply:
Actually I'll move some logic to willUpdate hook, stay tuned.